---
title: "ATAC-seq data"
author: Heejung Shim
output: html_document
---

Here, we will show how to obtain ATAC-seq data on chr1:568739-569762 for three copper treated samples and three media control samples and do data-preprocessing as described Shim et al 2021. 

First download hdf5 files for all samples from [this link](https://melbourne.figshare.com/articles/dataset/ATAC-seq_data_from_Shim_et_al_2021/14748480) and save them in the following directory:
```{}
cd multiseq_Shim_et_al/ATACseq/information_results/
```

```{r}
## path to `multiseq_Shim_et_al' repository on your computer
path.multiseq_Shim_et_al ="xxx"

## set work directory 
setwd(path.multiseq_Shim_et_al)

## Path to directory which contain ATAC-seq data as hdf5 format, 
hdf5.data.path = paste0(path.multiseq_Shim_et_al, "ATACseq/information_results/")

## load the R scripts to read data from hdf5 files and handle pcr artifacts 
source("scripts/utils.R")

## name for each condition: N702 for copper treatment, N706 for control 1 (media control), and N705 for control 2 (ethanol control)
name.control = "N706"
name.treatment = "N702"

## sample names
names.Sam = c("N501", "N502", "N503")

## Make a list of sample names and a list of hdf5 file names : treatment first and control later.
sample.names = c(paste0(name.treatment, names.Sam), paste0(name.control, names.Sam))
sample.files = paste0(sample.names, ".qfiltered10")

## location information 
chr= 1
st.posi =  568739
en.posi =  569762

numSam = length(sample.names)
numBPs = en.posi - st.posi + 1
```

We will read data and do data-preprocessing (removing reads that are potentially artifacts due to PCR amplication).
```{r}
## save the ATAC-seq data
ATAC.dat = matrix(data=0, nr = numSam, nc = numBPs)

## save positions with pcr artifacts for each strand
pcr.posi = vector("list", 2)

## for fwd strand
## read read counts for a given region
## for + strand, we need get reads at locations that are shifted 4bp to left
ATAC.dat.fwd = matrix(data=NA, nr = numSam, nc = numBPs)
for(i in 1:numSam){
  path.fwd = paste0(hdf5.data.path, sample.files[i] , ".fwd.h5")
  ATAC.dat.fwd[i, 1:numBPs] = as.matrix(get.counts.h5(path.fwd, paste0("chr", chr), st.posi-4, en.posi-4))
}

## remove pcr artifacts
pcr.ix = 1
pcr.removed.fwd = remove.pcr.artifacts(data=ATAC.dat.fwd, win.half.size=50, prop.thresh=0.9)
ATAC.dat = ATAC.dat + pcr.removed.fwd$data
if(!is.null(pcr.removed.fwd$posi.with.pcr.artifacts)){
  pcr.posi[[pcr.ix]] = pcr.removed.fwd$posi.with.pcr.artifacts
}

## for reverse
## read read counts for a given region
## for - strand, we need get reads at locations that are shifted 4bp to right
ATAC.dat.rev = matrix(data=NA, nr = numSam, nc = numBPs)
for(i in 1:numSam){
  path.rev = paste0(hdf5.data.path, sample.files[i] , ".rev.h5")
  ATAC.dat.rev[i, 1:numBPs] = as.matrix(get.counts.h5(path.rev, paste0("chr", chr), st.posi+4, en.posi+4))
}

pcr.ix = 2
pcr.removed.rev = remove.pcr.artifacts(data=ATAC.dat.rev, win.half.size=50, prop.thresh=0.9)
ATAC.dat = ATAC.dat + pcr.removed.rev$data
if(!is.null(pcr.removed.rev$posi.with.pcr.artifacts)){
    pcr.posi[[pcr.ix]] = pcr.removed.rev$posi.with.pcr.artifacts
}
```

We will match the sequencing depth by subsampling reads. This step is only required when we build the empirical distribution of the likelihood ratio test statistic by using an extra data set that is expected to have no difference in phenotype among samples (e.g., our ATAC-seq analysis in the main text).
```{r}
## Read sampling probabilities
sampling.prob = scan(paste0(hdf5.data.path, "Copper.prob"))

## sample data - feel free to change seed number
set.seed(789)
phenoD = matrix(data=0, nr = numSam, nc = numBPs)

for(i in 1:numSam){
  phenoD[i,] = rbinom(numBPs, ATAC.dat[i,], sampling.prob[i])
}
```
`phenoD` contains 6 by 1024 read counts for 6 samples (3 copper treated vs 3 media control) on chr1:568739-569762.
